array/zfp/pointer1.h

```diff
@@ -21,10 +22,18 @@ class pointer {
   friend class array1;
   friend class reference;
   explicit pointer(reference r) : ref(r) {}
+public:
   explicit pointer(array1* array, uint i) : ref(array, i) {}
+protected:
   ptrdiff_t index() const { return ref.i; }
   void set(ptrdiff_t index) { ref.i = index; }
   void increment() { ref.i++; }
   void decrement() { ref.i--; }
   reference ref;
+
+public:
```

Comment:
More C++11 stuff.  Does RAJA need the type alias?  I want to make sure dims and size() are needed.  It is somewhat unnatural to ask for a pointer's size and have it return the size of the array into which is points.  There's no equivalent operator for native pointers.

Reply:
I removed reference_t and dims.

However, I added data() which returns the underlying array. It is needed to support View1D (I need to know the size of the data contained by the underlying ZFP array).

Reply:
I'm still not convinced that a `data()` method is needed.  You should think of `zfp::array<double>::pointer` as a proxy for `double*`, and these types should in principle have identical semantics.  When initializing `double* ptr = &my_stl_vector[0];`, there's no way to query how much storage was allocated for (or any other information about) the vector via `ptr` alone (i.e., this information is not implicit).  Why is such a method needed for zfp arrays but not for uncompressed arrays (e.g., STL vectors) in the context of Kripke/Raja?

I'm not trying to make life more difficult for you, but I'm hesitant to add public API functions (plus documentation and tests) that seem nonessential and that may not even be implementable, e.g., if the `pointer` implementation one day changes in a way that precludes access to the underlying array.  I suspect there's a cleaner way of supporting what you want to do.  If you want zfp's pointer to behave more like a view, then I would suggest either using zfp's views or deriving your own view from zfp's pointer.

Reply:
I only need this method in the short term. I believe I will be able to obtain that information directly in RAJA::View.